File: Problem1.py
#find missing element from array

def findMissingElement(nums):
  start, end = 0, len(nums)-1
  while start <= end:
    mid = start + (end - start) // 2
    if nums[mid] - mid != nums[start] - start:
      end = mid
    elif nums[end] - end != nums[mid] - mid:
      start = mid
    if end - start <= 1:
      return nums[start]+1
    

x = findMissingElement([1,2,3,5,6,7,8,9])
print(x)

File: Problem1.java
//find missing element in array
class Main {
  public static void main(String[] args) {
    int[] arr = new int[] {2,4};
    System.out.println(getMin(arr));
  }

  public static int getMin(int[] nums){
    int start = 0;
    int end = nums.length - 1;

    while(end - start >= 1){
      int mid = start + (end - start) /2;
      if(nums[mid]-mid != nums[start]-start){
        end = mid;
      }
      else if(nums[mid] - mid != nums[end] - end){
        start = mid;
      }
      if(end-start == 1){
        return nums[start]+1;
      }
    }
    return -1;
  }
}
